### ISSUE TITLE
Timetone Conversion Mismatch between Editor and Schedule when "Now" and "Event" are in different Daylight Saving Times

### ISSUE THREAD

**Issue by MaZderMind** [pretalx/pretalx#1444]
### Problem and impact

We have an Event in US/Pacific.

On "the second Sunday in March" which this year was 2023-03-12, at least most of US/Pacific switched from PST which is UTC−08:00 to PDT (UTC−07:00).

Today is 2023-03-13.

Here in Germany we still have Winter Time (CET = UTC+1:00) and will switch to Summer Time (CEST = UTC+2:00) on 2023-03-26.

Within Pretalx we have Talkslots saved as follows:
```
pretalx=# select * from schedule_talkslot where room_id = 81;
  id   |         start          |          end           | room_id | schedule_id | submission_id | is_visible | description |            updated            
-------+------------------------+------------------------+---------+-------------+---------------+------------+-------------+-------------------------------
 16824 | 2023-04-17 18:00:00+00 | 2023-04-17 18:50:00+00 |      81 |         345 |           925 | t          |             | 2023-03-13 12:29:17.145893+00
```

Submission ID `925` corresponds to `Registration and Breakfast`.

In the Schedule View it is displayed as `11:00 AM`:
<img width="511" alt="Screenshot 2023-03-13 at 14 19 54" src="https://user-images.githubusercontent.com/142237/224713885-98c9a2cc-a5ba-4a12-87e1-a8c04ed8ac13.png">

However in the Editor it is displayes as `10:00 AM`

<img width="845" alt="Screenshot 2023-03-13 at 14 21 29" src="https://user-images.githubusercontent.com/142237/224714075-4da78a60-8187-4d97-8f0d-78ee2bb86054.png">

despite both showing `US/Pacific` as the selected Timezone.

As far as I can tell, the Schedule View is correct by using UTC-7:00 to make Local 11:00 from UTC 18:00, however the Editor seems to to it incorrectly.

### Expected behaviour

Same Time displayed in Editor and View, coherent with Time saved in the DB

**Comment by MaZderMind** [comment]
My best guess currently is a mismatch in how the JS-Side and the Py-Side treat the timezone conversion; this is the direction I'm continuing to debug.

**Comment by MaZderMind** [comment]
That seems kind-of confirmed and might be a bug originating in an old or somehow broken momentjs version being in use.

In the Chrome-Debugger on our Pretalx-Instance the following can be observerd:
```
moment.tz('2023-04-17T18:00:00+00', 'US/Pacific').format("HH:mm")
'10:00'
```

While the same Code being run on the momentjs homepage results in this output:
```
moment.tz('2023-04-17T18:00:00+00', 'US/Pacific').format("HH:mm")
'11:00'
```

However, the `moment-timezone-with-data-10-year-range.js` is, according to the header-comment in 0.5.40 which is only one minor release old.

**Comment by MaZderMind** [comment]
Ah, `[moment-with-locales.js](https://github.com/pretalx/pretalx/blob/main/src/pretalx/static/vendored/moment-with-locales.js)` is ~ 6 years old, so there might have been a bug or two fixed since back then.

**Comment by rixx** [comment]
Closed six weeks ago in 696a148d. (Which is, as a side note, why there's a "version" field in the bug reporting template.)

**Comment by rixx** [comment]
ref #1421
